src/app-router.js

```diff
@@ -256,7 +264,11 @@ function resolveInstruction(instruction, result, isInnerInstruction, router) {
 function restorePreviousLocation(router) {
   let previousLocation = router.history.previousLocation;
   if (previousLocation) {
-    router.navigate(router.history.previousLocation, { trigger: false, replace: true });
+    Promise.resolve().then(() => {
+      if (router.lastHistoryMovement && !isNaN(router.lastHistoryMovement)) {
+        router.history.history.go(-router.lastHistoryMovement);
```

Comment:
Does this read "assume the History implementation always have a history pointing to underlying platform History"? @jwx

Reply:
Err, yeah, unfortunately. It should be rewritten so that it only goes to the `History` implementation (which already requires `navigateBack` so `go` or `move` should be acceptable there). Once @davismj has taken a look at this "preliminary" PR and approved of my solution to the problem, I'll fix it.